### PR TITLE
AV 1056 implement gas sensor

### PR DESCRIPTION
**Pull request by SuragNuthulapaty**
added and implemented gas sensor stuff based on TARS

### CODE REVIEW

File: MIDAS/src/systems.cpp [hunk 1]
```diff
@@ -68,6 +68,7 @@ DECLARE_THREAD(gps, RocketSystems* arg) {
 
 DECLARE_THREAD(gas, RocketSystems* arg) {
     while (true) {
+        arg->rocket_state.gas.update(arg->sensors.gas.read());
         THREAD_SLEEP(16);
```
Comment (aidancostello) [inline MIDAS/src/systems.cpp:72]: change the thread sleep to as closely match the maximum sampling rate as possible

File: MIDAS/src/systems.cpp [hunk 2]
```diff
@@ -68,7 +68,8 @@ DECLARE_THREAD(gps, RocketSystems* arg) {
 
 DECLARE_THREAD(gas, RocketSystems* arg) {
     while (true) {
-        THREAD_SLEEP(16);
+        arg->rocket_state.gas.update(arg->sensors.gas.read());
+        THREAD_SLEEP(5.5);
```
Comment (redindelible) [inline MIDAS/src/systems.cpp:72]: Does the fractional tick work?

**Review by redindelible — APPROVED** [review]

File: MIDAS/src/hardware/Gas.cpp [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        bme.beginReading();
+    } else if (remaining == 0) {
+        bme.performReading();
+        temperature = bme.temperature;
+        humidity = bme.humidity;
+        pressure = bme.pressure;
+        resistance = bme.gas_resistance;
+    }
+    
+    return {.humidity = humidity, .temperature = temperature, .pressure = pressure, .resistance = resistance};
```
Comment (aidancostello) [inline MIDAS/src/hardware/Gas.cpp:42]: this line doesn't build in platformio, something with the way that this is returned. you can maybe try { humidity, temperature, pressure, resistance};

**Review by aidancostello — CHANGES_REQUESTED** [review]
You also have to init the sensor in init_sensors() in systems.cpp

**Comment by redindelible** [comment]
Merge with master plz.

**Review by aidancostello — APPROVED** [review]
looks good, don't merge just yet

**Review by aidancostello — APPROVED** [review]

**Pull request merged**